File: typescript/triangle/triangle.ts
export default class Triangle {
	sides: number[];

	constructor(...sides: number[]) {
		this.sides = sides;
	}

	kind(): string {
		const a: number = this.sides[0];
		const b: number = this.sides[1];
		const c: number = this.sides[2];
		if (
			this.sides.length != 3 ||
			this.sides.some(s => s <= 0) ||
			a + b < c ||
			c + b < a ||
			a + c < b
		) {
			throw new Error('Invalid sides');
		}
		if (a === b && a === c) return 'equilateral';
		if (a === b || a === c || b === c) return 'isosceles';
		return 'scalene';
	}
}

File: python/triangle/triangle.py
def equilateral(sides):
    return len({s for s in sides}) == 1 and sides[0] != 0


def isosceles(sides):
    return len({s for s in sides}) <= 2 and all(s != 0 and sum(sides) - 2*s > 0 for s in sides)


def scalene(sides):
    return len({s for s in sides}) == 3 and all(s != 0 and sum(sides) - 2*s > 0 for s in sides)
